activestorage/lib/active_storage/service.rb

```diff
@@ -152,6 +152,23 @@ def inspect # :nodoc:
       "#<#{self.class}#{name.present? ? " name=#{name.inspect}" : ""}>"
     end
 
+    def compute_checksum(io, **)
+      raise ArgumentError, "io must be rewindable" unless io.respond_to?(:rewind)
+
+      checksum_implementation.new.tap do |checksum|
+        read_buffer = "".b
+        while io.read(5.megabytes, read_buffer)
```

Comment:
I researched the chunking topic. All sharing algorithms (md5, sha-2/3 etc) are shaving o(n) complexity, so chunking does give any performance benefit if not performed in parallel.

I am not sure why chunking appeared in the first place, but it will not give any performance benefit in the current form.

Reply:
Rails has already been doing 5.megabyte chunks or deferring to the `file` implementation of the Digest class. I also do not think it would make sense to remove it.

I added a `default_chunk_size` configuration to ActiveStorage for the existing value (and added a case to disable it with `.to_i == 0`). Service classes could respect this or implement their own, perhaps tuned to the algorithm in use.

For new vs old data, the chunk size does not effect the calculation of the checksum value for MD5 so persisted data/checksums would continue to function with a changed configuration setting. Other algorithms may vary, but that would be the responsibility of the service classes that implement them.